src/main/io/vtx_rtc6705.c

```diff
@@ -0,0 +1,325 @@
+
+    static bool configured = false;
+    if (!configured) {
+        rtc6705Dev.band = vtxRTC6705Config()->band;
+        rtc6705Dev.channel = vtxRTC6705Config()->channel;
+        rtc6705Dev.power = vtxRTC6705Config()->power;
+
+#ifdef RTC6705_POWER_PIN
+        if (rtc6705Dev.power > 0) {
+            vtxRTC6705EnableAndConfigure();
+        } else {
+            rtc6705Disable();
+        }
+#else
+        vtxRTC6705Configure();
+#endif
+
+        configured = true;
+    }
+}
+
+#ifdef VTX_COMMON
+// Interface to common VTX API
+
+vtxDevType_e vtxRTC6705GetDeviceType(void)
+{
+    return VTXDEV_RTC6705;
+}
+
+bool vtxRTC6705IsReady(void)
+{
+    return true;
+}
+
+void vtxRTC6705SetBandChan(uint8_t band, uint8_t channel)
+{
+    WAIT_FOR_VTX;
+
+    if (band && channel) {
+        if (rtc6705Dev.power > 0) {
+            rtc6705SetChannel(band - 1, channel - 1);
+        }
+
+        rtc6705Dev.band = band;
+        rtc6705Dev.channel = channel;
+
+        vtxRTC6705ConfigMutable()->band = band;
+        vtxRTC6705ConfigMutable()->channel = channel;
+    }
+}
+
+void vtxRTC6705SetPowerByIndex(uint8_t index)
+{
+    WAIT_FOR_VTX;
+
+    vtxRTC6705ConfigMutable()->power = index;
+
+#ifdef RTC6705_POWER_PIN
+    if (index == 0) {
+        // power device off
+
+        if (rtc6705Dev.power > 0) {
+            // on, power it off
+            rtc6705Dev.power = index;
+            rtc6705Disable();
+            return;
+        } else {
+            // already off
+        }
+    } else {
+        // change rf power and maybe turn the device on first
+        if (rtc6705Dev.power == 0) {
+            // if it's powered down, power it up, wait and configure channel, band and power.
+            rtc6705Dev.power = index;
+            vtxRTC6705EnableAndConfigure();
+            return;
+        } else {
+            // if it's powered up, just set the rf power
+            rtc6705Dev.power = index;
+            rtc6705SetRFPower(index);
+        }
+    }
+#else
+    rtc6705Dev.power = index;
+    rtc6705SetRFPower(index);
+#endif
+}
+
+void vtxRTC6705SetPitmode(uint8_t onoff)
+{
+    UNUSED(onoff);
+    return;
+}
+
+bool vtxRTC6705GetBandChan(uint8_t *pBand, uint8_t *pChan)
+{
+    *pBand = rtc6705Dev.band;
+    *pChan = rtc6705Dev.channel;
+    return true;
+}
+
+bool vtxRTC6705GetPowerIndex(uint8_t *pIndex)
+{
+    *pIndex = rtc6705Dev.power;
+    return true;
+}
+
+bool vtxRTC6705GetPitmode(uint8_t *pOnOff)
+{
+    UNUSED(pOnOff);
+    return false;
+}
+
+static vtxVTable_t rtc6705VTable = {
+    .process = vtxRTC6705Process,
+    .getDeviceType = vtxRTC6705GetDeviceType,
+    .isReady = vtxRTC6705IsReady,
+    .setBandChan = vtxRTC6705SetBandChan,
+    .setPowerByIndex = vtxRTC6705SetPowerByIndex,
+    .setPitmode = vtxRTC6705SetPitmode,
+    .getBandChan = vtxRTC6705GetBandChan,
+    .getPowerIndex = vtxRTC6705GetPowerIndex,
+    .getPitmode = vtxRTC6705GetPitmode,
+};
+#endif // VTX_COMMON
+
+#ifdef CMS
+
+static uint8_t cmsx_vtxBand;
```

Comment:
That's reasonable. It's probably better to address these together in a later PR.